app/scripts/models/user.js

```diff
@@ -342,6 +343,14 @@ define(function (require, exports, module) {
       return account.signUp(password, relier, options)
         .then(function () {
           return self.setSignedInAccount(account);
+        }, function (err) {
+          if (! AuthErrors.is(err, 'ACCOUNT_ALREADY_EXISTS')) {
+            throw err;
```

Comment:
Maybe put line 353 inside the branch, invert the condition to remove the `!`, and throw any remaining errors on line 353.